src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/service/ReferralServiceTest.kt

```diff
@@ -1,57 +1,71 @@
-
 import org.assertj.core.api.Assertions.assertThat
 import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest
+import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto.DraftReferral
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Referral
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.ReferralRepository
 import java.time.LocalDate
 import java.time.OffsetDateTime
-import java.util.UUID
-
-class ReferralServiceTest {
-  private fun newOriginal(): Referral {
-    return Referral(
-      id = UUID.fromString("3529d30e-3480-4686-b14b-485beb3431ed"),
-      created = OffsetDateTime.parse("2021-05-05T13:00:01+01:00"),
-      // 👇 business fields
-      completionDeadline = LocalDate.parse("2021-06-26")
-    )
-  }
+
+@DataJpaTest
+class ReferralServiceTest @Autowired constructor(
+  val entityManager: TestEntityManager,
+  val referralRepository: ReferralRepository
+) {
+
+  private val referralService = ReferralService(referralRepository)
 
   @Test
   fun `update cannot overwrite identifier fields`() {
-    val update = DraftReferral(
-      id = UUID.fromString("ce364949-7301-497b-894d-130f34a98bff"),
+    val referral = Referral()
+    entityManager.persist(referral)
+    entityManager.flush()
+
+    val draftReferral = DraftReferral(
       created = OffsetDateTime.parse("2020-12-01T00:00:00+00:00")
     )
 
-    val updated = ReferralService.updateReferral(newOriginal(), update)
-    assertThat(updated.id).isEqualTo(UUID.fromString("3529d30e-3480-4686-b14b-485beb3431ed"))
-    assertThat(updated.created).isEqualTo(OffsetDateTime.parse("2021-05-05T13:00:01+01:00"))
+    val updated = referralService.updateReferral(referral.id!!, draftReferral)
+    assertThat(updated!!.id).isEqualTo(referral.id!!)
+    assertThat(updated.created).isEqualTo(referral.created)
   }
 
   @Test
   fun `null fields in the update do not overwrite original fields`() {
-    val update = DraftReferral(completionDeadline = null)
+    val referral = Referral(completionDeadline = LocalDate.of(2021, 6, 26))
+    entityManager.persist(referral)
+    entityManager.flush()
+
+    val draftReferral = DraftReferral(completionDeadline = null)
 
-    val updated = ReferralService.updateReferral(newOriginal(), update)
-    assertThat(updated.completionDeadline).isEqualTo(LocalDate.parse("2021-06-26"))
+    val updated = referralService.updateReferral(referral.id!!, draftReferral)
+    assertThat(updated!!.completionDeadline).isEqualTo(LocalDate.parse("2021-06-26"))
   }
 
   @Test
   fun `non-null fields in the update overwrite original fields`() {
-    val update = DraftReferral(completionDeadline = LocalDate.parse("2020-12-01"))
+    val referral = Referral(completionDeadline = LocalDate.of(2021, 6, 26))
```

Comment:
nitpick (not a blocker): would be good to stick with one or other of `LocalDate.of` or `LocalDate.parse`

Reply:
You're right.  I prefer the LocalDate.of method, updated all.